nautobot/extras/migrations/0055_collect_roles_from_related_apps_roles.py

```diff
@@ -0,0 +1,90 @@
+# Generated by Django 3.2.16 on 2022-11-19 23:15
+
+from collections import namedtuple
+from django.db import migrations, models
+
+from nautobot.ipam.choices import IPAddressRoleChoices
+
+
+# Role Models or RoleChoices that would be integrated into this Role model are referred to as RelatedRoleModelOrChoices.
+# For Example: DeviceRole, RackRole, IPAddressRoleChoices e.t.c.
+# while implemented_by refers to the model that actually implements the Related Model/Choice.
+# e.g role = models.ForeignKey(to=DeviceRole, ...)
+RelatedRoleModelOrChoices = namedtuple("RelatedRoleModelOrChoices", ["model_or_choices", "implemented_by"])
+
+# Using this, Choices is made to resemble a Role Queryset.
+ChoicesQuerySet = namedtuple("FakeQuerySet", ["name", "slug", "description", "color"])
+
+
+def populate_roles_from_related_app_roles(apps, schema_editor):
+    """Populate Role models using records from other related role models or choices from different apps."""
+
+    Role = apps.get_model("extras", "Role")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    related_role_models_or_choices = (
+        RelatedRoleModelOrChoices("dcim.DeviceRole", ["dcim.device", "virtualization.virtualmachine"]),
```

Comment:
The migration of DeviceRole should refer to the `vm_role` flag on each record to determine whether that particular Role should be assigned to `virtualization.virtualmachine`.